SteamKit2/SteamKit2/Steam/SteamClient/SteamClient.cs

```diff
@@ -342,5 +368,14 @@ void HandleServerList( IPacketMsg packetMsg )
             PostCallback( new ServerListCallback( listMsg.Body ) );
         }
 
+        void HandleJobHeartbeat( IPacketMsg packetMsg )
+        {
+            jobManager.HeartbeatJob( packetMsg.TargetJobID );
+        }
+        void HandleJobFailed( IPacketMsg packetMsg )
```

Comment:
It might be a good idea to turn this into a callback too, so that users not using `async` can see catch their failed jobs.